syntax = "proto3";

package mom;

message SavePendingServiceParameters {
  string task_id = 3;
  string client_id = 1;
  string service = 2;
  string payload = 4;
}

message SavePendingServiceResponse {
  string status = 1;
  string response = 2;
  string timestamp = 3;
}



message RetrievePendingServiceParameters {
  string task_id = 3;
  string client_id = 1;
}


message RetrievePendingServiceResponse {
  string status = 1;
  string response = 2;
  string timestamp = 3;
}



service MOMService {
  rpc SavePendingService(SavePendingServiceParameters) returns (SavePendingServiceResponse);
  rpc RetrievePendingService(RetrievePendingServiceParameters) returns (RetrievePendingServiceResponse);
}
